### ISSUE TITLE
Requests to /cloudfoundryapplication possibly return results for /actuator

### ISSUE THREAD

**Issue by TimHess** [SteeltoeOSS/Steeltoe#1293]
## Describe the bug
In CF context, only /health and /info are reported as being enabled

Sample response:
```json
{
    "type": "steeltoe",
    "_links": {
        "info": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication/info",
            "templated": false
        },
        "health": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication/health",
            "templated": false
        },
        "self": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication",
            "templated": false
        }
    }
}
```

## Steps to reproduce 
1. Add Actuators to app
2. Deploy to CF
3. View app in apps manager

## Expected behavior
All endpoints supported on Cloud Foundry should be returned/enabled automatically

## Environment (please complete the following information):
 - Steeltoe Version `main`
 - Platform: Cloud Foundry
 - OS: Linux
 - .NET Version 8.0

**Comment by TimHess** [comment]
Probably related:
```
2024-05-29T13:57:16.458-05:00 [APP/PROC/WEB/0] [OUT] IDX10242: Security token: '[Security Artifact of type 'Microsoft.IdentityModel.JsonWebTokens.JsonWebToken' is hidden. For more details, see https://aka.ms/IdentityModel/SecurityArtifactLogging.]' has a valid signature.
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] IDX10239: Lifetime of the token is valid.
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] IDX10214: Audience validation failed. Audiences: 'cloud_controller, actuator, apps_manager_js'. Did not match: validationParameters.ValidAudience: 'null' or validationParameters.ValidAudiences: 'sampleapi, 4e6f8e34-f42b-440e-a042-f2b13c1d5bed'.
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] info: Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler[1]
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] Failed to validate the token.
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] Microsoft.IdentityModel.Tokens.SecurityTokenInvalidAudienceException: IDX10214: Audience validation failed. Audiences: 'cloud_controller, actuator, apps_manager_js'. Did not match: validationParameters.ValidAudience: 'null' or validationParameters.ValidAudiences: 'sampleapi, 4e6f8e34-f42b-440e-a042-f2b13c1d5bed'.
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.Tokens.Validators.ValidateAudience(IEnumerable`1 audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateTokenPayloadAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] info: Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler[7]
```

**Comment by bart-vmware** [comment]
While looking into this, we should take the following [Sonar security hotspot](https://sonarcloud.io/project/security_hotspots?id=SteeltoeOSS_steeltoe&tab=code) into account:
![image](https://github.com/user-attachments/assets/9ddc60b3-71ff-4758-8206-fb196e2a47a5)

The proposal is to replace the wildcard with the hostname obtained from `VCAP_APPLICATION`. If unavailable, Steeltoe shouldn't set the wildcard policy.

**Comment by bart-vmware** [comment]
Once this is implemented, the test at `src/Management/test/Endpoint.Test/ActuatorsHostBuilderTest.cs` should be updated as described at https://github.com/SteeltoeOSS/Steeltoe/pull/1386#discussion_r1806846004.

**Comment by TimHess** [comment]
> Probably related:
> 
> ```
> 2024-05-29T13:57:16.458-05:00 [APP/PROC/WEB/0] [OUT] IDX10242: Security token: '[Security Artifact of type 'Microsoft.IdentityModel.JsonWebTokens.JsonWebToken' is hidden. For more details, see https://aka.ms/IdentityModel/SecurityArtifactLogging.]' has a valid signature.
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] IDX10239: Lifetime of the token is valid.
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] IDX10214: Audience validation failed. Audiences: 'cloud_controller, actuator, apps_manager_js'. Did not match: validationParameters.ValidAudience: 'null' or validationParameters.ValidAudiences: 'sampleapi, 4e6f8e34-f42b-440e-a042-f2b13c1d5bed'.
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] info: Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler[1]
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] Failed to validate the token.
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] Microsoft.IdentityModel.Tokens.SecurityTokenInvalidAudienceException: IDX10214: Audience validation failed. Audiences: 'cloud_controller, actuator, apps_manager_js'. Did not match: validationParameters.ValidAudience: 'null' or validationParameters.ValidAudiences: 'sampleapi, 4e6f8e34-f42b-440e-a042-f2b13c1d5bed'.
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.Tokens.Validators.ValidateAudience(IEnumerable`1 audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateTokenPayloadAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
> 2024-05-29T13:57:16.459-05:00 [APP/PROC/WEB/0] [OUT] info: Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler[7]
> ```

I'm not seeing evidence this was actually related. There was no special handling present for `/cloudfoundryapplication` in the current version of the exposure logic for actuator endpoints. In retrospect, I think these log entries were more likely due to something specific to that sample application

**Comment by TimHess** [comment]
> While looking into this, we should take the following [Sonar security hotspot](https://sonarcloud.io/project/security_hotspots?id=SteeltoeOSS_steeltoe&tab=code) into account: ![image](https://private-user-images.githubusercontent.com/104792814/362336776-9ddc60b3-71ff-4758-8206-fb196e2a47a5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyOTc4NDYsIm5iZiI6MTczMDI5NzU0NiwicGF0aCI6Ii8xMDQ3OTI4MTQvMzYyMzM2Nzc2LTlkZGM2MGIzLTcxZmYtNDc1OC04MjA2LWZiMTk2ZTJhNDdhNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxNDEyMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDFkMGQzY2NkZDY4NTE3ZTkzYWQ5YjhiZmI5MWFlNWMxZWM4MTE3YmQwNWUzNWY5MWVlMDI0NzU3M2ZkMWI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.R0oVVfMWM7L9n5U4IOsDGs_0b6Be69WEwO7PdbF_C0c)
> 
> The proposal is to replace the wildcard with the hostname obtained from `VCAP_APPLICATION`. If unavailable, Steeltoe shouldn't set the wildcard policy.
> 
> Once implemented, #959 can be closed.

I looked at this again this morning, but I don't think there's a way to do this. The URIs provided by VCAP_APPLICATION will match `app-route.apps-domain`, but Apps Manager will be running under `apps.system-domain`, where (I think) apps-domain and system-domain can't match exactly... Even in cases (like our test environments) where they are close, we'd need to backtrack from the full route to a guess at what apps-domain probably is, and guess at where apps manager lives. We're better off improving the documentation for how to do this precisely.
